internal/repository/payment.go

```diff
@@ -47,13 +48,51 @@ func (r *PaymentRepository) GetAllPaymentIntents() []model.Payment {
 	return payments
 }
 
+func (r *PaymentRepository) GetModePaymentIntents(mode enum.Mode) []model.Payment {
+	var payments []model.Payment
+	r.DB.
+		Preload("Account").
+		Where("mode = ?", mode.String()).
+		Preload("CurrentPaymentState").
+		Preload("PaymentStates").
+		Joins("CurrentPaymentState").
+		Where("\"CurrentPaymentState\".\"status_name\" IN ?", []string{"waiting", "partially_paid"}).
+		Find(&payments)
+	return payments
+}
+
+func (r *PaymentRepository) GetPaymentIntentByAddress(address *common.Address) (*gorm.DB, model.Payment) {
+	var payment model.Payment
+	result := r.DB.Debug().
+		Preload("Account").
+		Preload("CurrentPaymentState").
+		Preload("PaymentStates").
+		Joins("Account").
+		Joins("CurrentPaymentState").
+		Where("\"CurrentPaymentState\".\"status_name\" IN ? AND \"Account\".\"address\" = ?", []string{"waiting", "partially_paid"}, address.Hex()).
+		First(&payment)
+	return result, payment
+}
+
 func (r *PaymentRepository) GetAllUnfinishedPayments() []model.Payment {
 	var payments []model.Payment
 	r.DB.
 		Preload("Account").
 		Preload("CurrentPaymentState").
 		Joins("CurrentPaymentState").
-		Where("\"CurrentPaymentState\".\"status_name\" IN ?", []string{enum.Paid.String(), enum.Forwarded.String()}).
+		Where("\"CurrentPaymentState\".\"status_name\" IN ?", []string{enum.Confirmed.String(), enum.Forwarded.String()}).
```

Comment:
ich han bi mir enums als zahl ide db. wönd mirs einheitlich mache in zuekunft und wenn ja weli variante?

Reply:
hmm ja mir chönd au zahl neh